### ISSUE TITLE
[TODO]: v2 cross-validation

### ISSUE THREAD

**Issue by kevingreenman** [chemprop/chemprop#524]
From @JacksonBurns:
- we need to rethink how cross validation works
- currently it does not at all, since it will return the same list of indices repeart num_folds times
- and then we don't actually have any support for using them in the training script anyway

**Comment by davidegraff** [comment]
It's likely that [this line](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/v2/data/utils.py#L68) is the culprit:
```python
astartes_kwargs = dict(train_size=sizes[0], test_size=sizes[2], return_indices=True, random_state=seed)
```
which resets the random state of each fold split to `seed`. That is, if we request `5` folds with `42` as a seed, we're essentially calling `split_fun(sampler="random")` 5 times with 42 as a seed so each fold would necessarily be the same.

**Comment by JacksonBurns** [comment]
Resolved in #610, also see #608
